src/main/java/com/google/api/codegen/transformer/go/GoGapicSurfaceTransformer.java

```diff
@@ -0,0 +1,247 @@
+import com.google.api.codegen.transformer.MethodTransformerContext;
+import com.google.api.codegen.transformer.ModelToViewTransformer;
+import com.google.api.codegen.transformer.ModelTypeTable;
+import com.google.api.codegen.transformer.PageStreamingTransformer;
+import com.google.api.codegen.transformer.PathTemplateTransformer;
+import com.google.api.codegen.transformer.SurfaceTransformerContext;
+import com.google.api.codegen.util.go.GoTypeTable;
+import com.google.api.codegen.viewmodel.RetryPairDefinitionView;
+import com.google.api.codegen.viewmodel.StaticLangApiMethodView;
+import com.google.api.codegen.viewmodel.StaticLangXCombinedView;
+import com.google.api.codegen.viewmodel.ViewModel;
+import com.google.api.gax.core.RetrySettings;
+import com.google.api.tools.framework.aspects.documentation.model.DocumentationUtil;
+import com.google.api.tools.framework.aspects.documentation.model.ElementDocumentationAttribute;
+import com.google.api.tools.framework.model.Interface;
+import com.google.api.tools.framework.model.Method;
+import com.google.api.tools.framework.model.Model;
+import com.google.api.tools.framework.model.ProtoElement;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
+import io.grpc.Status.Code;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.TreeSet;
+
+public class GoGapicSurfaceTransformer implements ModelToViewTransformer {
+
+  private static final String XAPI_TEMPLATE_FILENAME = "go/main.snip";
+  private static final String SAMPLE_TEMPLATE_FILENAME = "go/example.snip";
+  private static final String DOC_TEMPLATE_FILENAME = "go/doc.snip";
+
+  private final ApiCallableTransformer apiCallableTransformer = new ApiCallableTransformer();
+  private final ApiMethodTransformer apiMethodTransformer = new ApiMethodTransformer();
+  private final PageStreamingTransformer pageStreamingTransformer = new PageStreamingTransformer();
+  private final PathTemplateTransformer pathTemplateTransformer = new PathTemplateTransformer();
+
+  @Override
+  public List<String> getTemplateFileNames() {
+    return Arrays.asList(XAPI_TEMPLATE_FILENAME);
+  }
+
+  @Override
+  public List<ViewModel> transform(Model model, ApiConfig apiConfig) {
+    List<ViewModel> models = new ArrayList<ViewModel>();
+    for (Interface service : new InterfaceView().getElementIterable(model)) {
+      models.add(generate(model, service, apiConfig));
+    }
+    return models;
+  }
+
+  private StaticLangXCombinedView generate(Model model, Interface service, ApiConfig apiConfig) {
+    StaticLangXCombinedView.Builder view = StaticLangXCombinedView.newBuilder();
+
+    view.templateFileName(XAPI_TEMPLATE_FILENAME);
+    view.serviceDoc(comments(service));
+
+    GoSurfaceNamer namer = new GoSurfaceNamer(model, service, apiConfig);
```

Comment:
Go through the SurfaceNamer abstraction instead of a language-specific implementation. Have the generate method just take a SurfaceTransformerContext; put the new concepts you need from the SurfaceNamer into the SurfaceNamer class.